From 3ec410f720930d537b16d2b5897465df6376e3d7 Mon Sep 17 00:00:00 2001
Message-Id: <3ec410f720930d537b16d2b5897465df6376e3d7.1417780878.git.chang-joon.lee@intel.com>
In-Reply-To: <cb098d33cdac3271103834fbce08218248d7bddb.1417780878.git.chang-joon.lee@intel.com>
References: <cb098d33cdac3271103834fbce08218248d7bddb.1417780878.git.chang-joon.lee@intel.com>
From: Durgadoss R <durgadoss.r@intel.com>
Date: Tue, 11 Nov 2014 16:10:42 +0530
Subject: [PATCH 14/34] FOR_UPSTREAM [VPG]: drm/i915: Optimize _vdd_on calls

We need not call _vdd_on if we already have panel
power up. This saves time in link training sequence.
Also, we dont use aux irq's in valleyview based
platforms.

Issue: GMINL-3876
Change-Id: I2c6e6cf1d263748cefccd0370443635370ae6110
Signed-off-by: Durgadoss R <durgadoss.r@intel.com>
---
 drivers/gpu/drm/i915/intel_dp.c |    8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_dp.c b/drivers/gpu/drm/i915/intel_dp.c
index 19b873f..c768727 100644
--- a/drivers/gpu/drm/i915/intel_dp.c
+++ b/drivers/gpu/drm/i915/intel_dp.c
@@ -624,10 +624,12 @@ intel_dp_aux_ch(struct intel_dp *intel_dp,
 	int i, ret, recv_bytes;
 	uint32_t status;
 	int try, clock = 0;
-	bool has_aux_irq = HAS_AUX_IRQ(dev);
-	bool vdd;
+	bool has_aux_irq = HAS_AUX_IRQ(dev) && !IS_VALLEYVIEW(dev);
+	bool vdd = false;
 
-	vdd = _edp_panel_vdd_on(intel_dp);
+	/* If we already have panel power, do not call _vdd_on */
+	if (!edp_have_panel_power(intel_dp))
+		vdd = _edp_panel_vdd_on(intel_dp);
 
 	/* dp aux is extremely sensitive to irq latency, hence request the
 	 * lowest possible wakeup latency and so prevent the cpu from going into
-- 
1.7.9.5

